src/controller/socialLoginController.js

```diff
@@ -0,0 +1,146 @@
+          success: false,
+          message: 'You had started the registration process earlier. '
+          + 'Please check your email to complete your registration.'
+        });
+      }
+      if (userFound) {
+        const token = await Authentication.getToken(userFound);
+        return obj.res.status(200).json({
+          success: true,
+          message: `Social login via ${obj.socialMedia} was successful`,
+          id: userFound.id,
+          username: userFound.username,
+          email: userFound.email,
+          token,
+        });
+      }
+      return obj.res.status(400).json({
+        success: false,
+        message: 'You are not registered on this app. Please signup.',
+      });
+    } catch (error) {
+      if (error.errors) {
+        return helperMethods.sequelizeValidationError(obj.res, error);
+      }
+      return helperMethods.serverError(obj.res);
+    }
+  }
+
+  /**
+    * Sign up a user
+    * Route: GET: auth/social_fb/callback
+    * @param {object} req - Request object
+    * @param {object} res - Response object
+    * @return {res} res - Response object
+    * @memberof UserController
+  */
+  static async facebookLogin(req, res) {
+    const verifyWithObjectProperties = {
+      verifyWith: 'email',
+      profile: req.user,
+      socialMedia: 'facebook',
+      res,
+    };
+    try {
+      return SocialLogin.verifyAndLoginUser(verifyWithObjectProperties);
+    } catch (error) {
+      if (error.errors) {
+        return helperMethods.sequelizeValidationError(res, error);
+      }
+      return helperMethods.serverError(res);
+    }
+  }
+
+  /**
+  * Sign up a user
+  * Route: GET: auth/social_ggl/callback
+  * @param {object} req - Request object
+  * @param {object} res - Response object
+  * @return {res} res - Response object
+  * @memberof UserController
+  */
+  static async googleLogin(req, res) {
+    const verifyWithObjectProperties = {
+      verifyWith: 'email',
+      profile: req.user,
+      socialMedia: 'google',
+      res,
+    };
+    try {
+      return SocialLogin.verifyAndLoginUser(verifyWithObjectProperties);
+    } catch (error) {
+      if (error.errors) {
+        return helperMethods.sequelizeValidationError(res, error);
+      }
+      return helperMethods.serverError(res);
+    }
+  }
+
+  /**
+  * Sign up a user
+  * Route: GET: auth/social_tw/callback
+  * @param {object} req - Request object
+  * @param {object} res - Response object
+  * @return {res} res - Response object
+  * @memberof UserController
+  */
+  static async twitterLogin(req, res) {
+    const verifyWithObjectProperties = {
+      verifyWith: 'username',
+      profile: req.user,
+      socialMedia: 'twitter',
+      res,
+    };
+    try {
+      return SocialLogin.verifyAndLoginUser(verifyWithObjectProperties);
+    } catch (error) {
+      if (error.errors) {
+        return helperMethods.sequelizeValidationError(res, error);
+      }
+      return helperMethods.serverError(res);
+    }
+  }
```

Comment:
There are a lot of return functions from line 120 to line 126. You can reduce them to reduce code complexity.

Reply:
All return statements are necessary for the code to work as expected.